fix(tauri): stabilise Linux WebKit and desktop packaging

Ludy87 opened the pull request:
# Description of Changes

This pull request makes significant improvements to the GitHub Actions workflows for multi-OS releases and Tauri builds. The main changes include enhanced job concurrency management, expanded Linux packaging options, refined dependency installation logic for Ubuntu builds, and updates to Java and Node.js setup steps. These updates aim to optimize CI resource usage, improve build reliability, and support more Linux distribution formats.

**Workflow and CI Improvements**

* Added `concurrency` configuration to both `.github/workflows/multiOSReleases.yml` and `.github/workflows/tauri-build.yml` to automatically cancel in-progress jobs when new jobs are triggered for the same workflow and branch, saving CI resources and time. [[1]](diffhunk://#diff-895b214ee023c8c26048a2a3b946cfb1ebc4f26fbc8a9c2fa54b77c12e763b6bR29-R40) [[2]](diffhunk://#diff-b34ab107dd4bc92075b2e89b6f16e4a2813e267ca7c2afebdb1931a0a3900d5aR26-R37)

**Linux Packaging and Build Matrix Expansion**

* Updated the build matrix logic in both workflows to support three separate Linux packaging formats (`AppImage`, `deb`, and `rpm`) for x86_64 Ubuntu builds, instead of only one previously. This enables broader distribution support. [[1]](diffhunk://#diff-895b214ee023c8c26048a2a3b946cfb1ebc4f26fbc8a9c2fa54b77c12e763b6bL77-R97) [[2]](diffhunk://#diff-b34ab107dd4bc92075b2e89b6f16e4a2813e267ca7c2afebdb1931a0a3900d5aL47-R72)

**Ubuntu Dependency Installation Logic**

* Split Ubuntu dependency installation into two jobs: one for general Ubuntu builds (excluding AppImage) and one for AppImage builds, with tailored package lists and direct `.deb` downloads for AppImage compatibility. Also added explanatory comments for each dependency. [[1]](diffhunk://#diff-895b214ee023c8c26048a2a3b946cfb1ebc4f26fbc8a9c2fa54b77c12e763b6bL171-R238) [[2]](diffhunk://#diff-b34ab107dd4bc92075b2e89b6f16e4a2813e267ca7c2afebdb1931a0a3900d5aL76-R142)

~**Java and Node.js Environment Updates**~

* ~Changed Java setup from JDK 21 to JDK 17 for both workflows, and switched Node.js dependency installation from `npm install` to `npm ci` for more reliable, reproducible builds. [[1]](diffhunk://#diff-895b214ee023c8c26048a2a3b946cfb1ebc4f26fbc8a9c2fa54b77c12e763b6bL114-R129) [[2]](diffhunk://#diff-895b214ee023c8c26048a2a3b946cfb1ebc4f26fbc8a9c2fa54b77c12e763b6bL190-R250) [[3]](diffhunk://#diff-895b214ee023c8c26048a2a3b946cfb1ebc4f26fbc8a9c2fa54b77c12e763b6bL129-R141) [[4]](diffhunk://#diff-b34ab107dd4bc92075b2e89b6f16e4a2813e267ca7c2afebdb1931a0a3900d5aL95-R174) [[5]](diffhunk://#diff-b34ab107dd4bc92075b2e89b6f16e4a2813e267ca7c2afebdb1931a0a3900d5aL176-R250)~

**AppImage-specific Build Enhancements**

* For AppImage builds, added steps to generate a temporary GPG key for signing, and to make `libjvm.so` discoverable by linking it into `/usr/lib`. Also adjusted signing logic for Windows builds and artifact upload patterns for more precise release asset matching. [[1]](diffhunk://#diff-b34ab107dd4bc92075b2e89b6f16e4a2813e267ca7c2afebdb1931a0a3900d5aR374-R389) [[2]](diffhunk://#diff-b34ab107dd4bc92075b2e89b6f16e4a2813e267ca7c2afebdb1931a0a3900d5aL310-R406) [[3]](diffhunk://#diff-b34ab107dd4bc92075b2e89b6f16e4a2813e267ca7c2afebdb1931a0a3900d5aL95-R174) [[4]](diffhunk://#diff-895b214ee023c8c26048a2a3b946cfb1ebc4f26fbc8a9c2fa54b77c12e763b6bL569-R630)

Let me know if you have questions about any of these changes or want to dig deeper into a specific part!

---

## Checklist

### General

- [ ] I have read the [Contribution Guidelines](https://github.com/Stirling-Tools/Stirling-PDF/blob/main/CONTRIBUTING.md)
- [ ] I have read the [Stirling-PDF Developer Guide](https://github.com/Stirling-Tools/Stirling-PDF/blob/main/devGuide/DeveloperGuide.md) (if applicable)
- [ ] I have read the [How to add new languages to Stirling-PDF](https://github.com/Stirling-Tools/Stirling-PDF/blob/main/devGuide/HowToAddNewLanguage.md) (if applicable)
- [ ] I have performed a self-review of my own code
- [ ] My changes generate no new warnings

### Documentation

- [ ] I have updated relevant docs on [Stirling-PDF's doc repo](https://github.com/Stirling-Tools/Stirling-Tools.github.io/blob/main/docs/) (if functionality has heavily changed)
- [ ] I have read the section [Add New Translation Tags](https://github.com/Stirling-Tools/Stirling-PDF/blob/main/devGuide/HowToAddNewLanguage.md#add-new-translation-tags) (for new translation tags only)

### Translations (if applicable)

- [ ] I ran [`scripts/counter_translation.py`](https://github.com/Stirling-Tools/Stirling-PDF/blob/main/docs/counter_translation.md)

### UI Changes (if applicable)

- [ ] Screenshots or videos demonstrating the UI changes are attached (e.g., as comments or direct attachments in the PR)

### Testing (if applicable)

- [ ] I have tested my changes locally. Refer to the [Testing Guide](https://github.com/Stirling-Tools/Stirling-PDF/blob/main/devGuide/DeveloperGuide.md#6-testing) for more details.

Ludy87 committed:
Add AppImage to Tauri bundle targets

Ludy87 committed:
Update Ubuntu dependencies in CI workflow

Added desktop-file-utils and libglib2.0-bin to the list of packages installed for Ubuntu-22.04 in the multiOSReleases workflow to ensure required tools are available for builds.

Ludy87 committed:
Update app version to 2.0.2 and fix workflow dependencies

Bump app version from 2.0.0 to 2.0.2 in tauri config and simulation configs. Add missing 'linuxdeploy' package and fix typo in GitHub Actions workflow dependencies for Ubuntu builds.

Ludy87 committed:
Update multiOSReleases.yml

Ludy87 committed:
Update multiOSReleases.yml

Ludy87 committed:
Refactor JRE build and dependencies in CI workflows

Ludy87 committed:
Update tauri-build.yml

Ludy87 committed:
Update tauri-build.yml

Ludy87 committed:
Update tauri-build.yml

Ludy87 committed:
Update tauri-build.yml

Ludy87 committed:
Update tauri.conf.json

Ludy87 committed:
Update tauri-build.yml

Ludy87 committed:
Update build workflow and bundle targets

Adds explicit build targets (appimage, deb, rpm) for Linux in the GitHub Actions workflow and expands Tauri bundle targets to include deb, rpm, dmg, app, and msi. Removes JRE caching step to always rebuild the custom JRE during CI.

Ludy87 committed:
Fix typo in tauri-build workflow args

Replaces '--buildes' with '--bundles' in the matrix configuration for Linux builds in the tauri-build GitHub Actions workflow to ensure correct argument usage.

Ludy87 committed:
Refactor Linux build steps in Tauri workflow

Linux build matrix no longer passes bundle arguments directly; instead, a dedicated step for building AppImage, deb, and rpm is added for the Ubuntu platform. Other platforms use the general Tauri build step, improving workflow clarity and separation of build logic.

Ludy87 committed:
Update tauri-build.yml

Ludy87 committed:
Add libglib2.0-dev to Ubuntu build dependencies

Included libglib2.0-dev in the list of packages installed for Ubuntu builds to ensure all required libraries are available for Tauri application compilation.

Ludy87 committed:
Update tauri-build.yml

Ludy87 committed:
Update tauri-build.yml

Ludy87 committed:
Update Tauri CLI and dependencies, adjust build workflow

Upgrades @tauri-apps/cli to v2.9.5 and updates related optional dependencies in package-lock.json. Also updates js-yaml and glob packages. The tauri-build GitHub workflow is modified to streamline and correct the list of required Ubuntu packages for builds.

Ludy87 committed:
Update tauri-build.yml

Ludy87 committed:
Update tauri-build.yml

Ludy87 committed:
Update tauri-build.yml

Ludy87 committed:
Add step to link libjvm for Ubuntu builds

Introduces a workflow step to create a symlink for libjvm.so on Ubuntu 22.04, ensuring the Java backend build can discover the JVM library. This addresses potential issues with library discovery during CI builds.

Ludy87 committed:
Update tauri build workflow and frontend dependencies

Modified tauri-build.yml to conditionally set APPIMAGETOOL_SIGN_PASSPHRASE only on main branch. Removed unused dev dependencies from frontend/package.json and package-lock.json, including madge and related packages.

Ludy87 committed:
Add --no-sign flag to Linux build in CI

Appends the --no-sign flag to the Linux build arguments in the GitHub Actions workflow to disable signing during CI builds for x86_64-unknown-linux-gnu targets.

Ludy87 committed:
Update tauri-build.yml

Ludy87 committed:
Update tauri-build.yml

Ludy87 committed:
Update tauri-build.yml

Ludy87 committed:
Update tauri-build.yml

Ludy87 committed:
Update tauri-build.yml

Ludy87 committed:
Update tauri-build.yml

Ludy87 committed:
Remove GPG key generation from build workflow

Removed temporary GPG key generation step for AppImage.

Ludy87 committed:
Generate GPG key for AppImage signing

Add step to generate temporary GPG key for AppImage signing on Ubuntu 22.04.

Ludy87 committed:
Refactor matrix generation for build platforms

Ludy87 committed:
Simplify matrix echo commands in tauri-build.yml

Ludy87 committed:
Update tauri-build.yml

Ludy87 committed:
Update tauri-build.yml

Ludy87 committed:
Remove license-checker and update Tauri config

Removed the 'license-checker' dev dependency from package.json and package-lock.json. Updated tauri.conf.json to explicitly include WebKit process binaries in the AppImage build. Also enabled Gradle cache in the GitHub Actions workflow and removed the AppImage functional test step.

Ludy87 committed:
Update tauri.conf.json

Ludy87 committed:
Update AppImage build to copy WebKit helper binaries

Adds a workflow step to copy WebKit2GTK helper binaries for AppImage builds on Ubuntu 22.04. Removes explicit file mapping from tauri.conf.json, relying on the workflow to provide required binaries.

Ludy87 committed:
Remove WebKit helper binaries copy step from CI

Eliminates the step that copies WebKit helper binaries for AppImage builds on Ubuntu 22.04 from the tauri-build GitHub Actions workflow. This may be due to changes in build requirements or improved packaging.

Ludy87 committed:
Update stirling-pdf.desktop

Ludy87 committed:
Enable software rendering for WebKit on Linux

Adds a function to set WEBKIT_DISABLE_COMPOSITING_MODE on Linux if not already set, enabling software rendering to avoid EGL issues. Logs a message when the environment variable is configured.

Ludy87 committed:
Update tauri-build.yml

Ludy87 committed:
Update tauri-build.yml

Ludy87 committed:
Improve Linux WebKit fallback configuration

Updated the GitHub Actions workflow to simplify package installation for WebKit dependencies on Ubuntu and removed explicit version pinning. Enhanced the Linux WebKit configuration in Rust to apply and log multiple environment variable fallbacks for better handling of EGL issues.

Ludy87 committed:
Update tauri-build.yml

Ludy87 committed:
Update tauri-build.yml

Ludy87 committed:
Update tauri-build.yml

Ludy87 committed:
Update tauri-build.yml

Ludy87 committed:
Update tauri-build.yml

Ludy87 committed:
Update tauri-build.yml

Ludy87 committed:
Update tauri-build.yml

Ludy87 committed:
Update tauri-build.yml

Ludy87 committed:
Update tauri-build.yml

Ludy87 committed:
Update tauri.conf.json

Ludy87 committed:
Update tauri-build.yml

Ludy87 committed:
Update tauri-build.yml

Ludy87 committed:
Simplify AppImage artifact processing in CI workflow

Removed FUSE installation and custom AppImage extraction, modification, and repackaging steps from the tauri-build GitHub Actions workflow. Now, AppImage files are simply copied to the dist directory, streamlining the build process for linux-x86_64-appimage targets.

Ludy87 committed:
Fix Tauri build workflow argument and platform handling

Uncomments the 'args' parameter for the Tauri build step to enable passing build arguments. Refines platform-specific artifact copying logic to handle AppImage files for Ubuntu and adds an explicit message for unknown platforms.

Ludy87 committed:
Update tauri.conf.json

Ludy87 committed:
Update Ubuntu dependency installation in CI workflow

Refines the dependency installation steps in the tauri-build GitHub Actions workflow. Adds conditional logic to separate AppImage-specific installs and updates the list of packages for Tauri v2 compatibility, including detailed comments explaining each dependency.

Ludy87 committed:
Restrict Ubuntu steps to AppImage build only

Updated workflow conditions so that making libjvm discoverable and generating a temporary GPG key only run for the linux-x86_64-appimage build on Ubuntu 22.04, instead of all Ubuntu builds.

Ludy87 committed:
Update CI workflows for improved Linux packaging and concurrency

Added concurrency control to multiOSReleases workflow to cancel in-progress jobs on new triggers. Expanded Linux build matrix to include AppImage, DEB, and RPM targets. Updated Ubuntu dependency installation for better compatibility with Tauri v2 and packaging formats. Changed JDK version from 21 to 17. Added step-security/harden-runner to tauri-build workflow for enhanced security.

Ludy87 committed:
Bump version to 2.0.3 and add GPG key generation for CI

Updated project version to 2.0.3 in build.gradle, tauri.conf.json, and test configs. Added a step in the multiOSReleases GitHub workflow to generate a temporary GPG key for AppImage signing on Ubuntu CI builds.

Frooodle commented on .github/workflows/multiOSReleases.yml:
```diff
@@ -111,10 +123,10 @@ jobs:
         uses: actions/setup-java@c5195efecf7bdfc987ee8bae7a71cb8b11521c00 # v4.7.1
         with:
-          java-version: "21"
+          java-version: "17"
```
why use jdk 17 not 21? did you have issues with 21?

Ludy87 committed:
Improve error handling and GPG key cleanup in CI workflows

Adds error handling to wget commands for package downloads in multiOSReleases.yml and tauri-build.yml to ensure the workflow fails on download errors. Updates GPG key generation to export the fingerprint and introduces a cleanup step to remove temporary GPG keys after use in tauri-build.yml.

Ludy87 committed:
Automate GPG key cleanup in AppImage build

Adds steps to export the GPG fingerprint to the environment and clean up the temporary GPG key after building the AppImage on Ubuntu. Also fixes heredoc usage for key generation parameters.

Ludy87 committed:
Update release workflows for RPM and signing logic

Added RPM package to multiOS release artifacts. Improved Windows signing logic in tauri-build workflow with detailed conditions for enabling signing.

Ludy87 committed:
Merge branch 'V2' into AppImage_fix

Ludy87 committed:
Update dependencies in package-lock.json

Added new optional dependencies including @emnapi/core, @emnapi/runtime, @emnapi/wasi-threads, and @napi-rs/canvas with platform-specific builds. Updated and expanded esbuild platform support packages and refreshed several ESLint-related dependencies.

Ludy87 committed:
Merge branch 'V2' into AppImage_fix

Ludy87 committed:
Merge branch 'main' into AppImage_fix

Ludy87 committed:
Update JDK version to 21 in CI workflow

Changed the Java setup steps in multiOSReleases.yml to use JDK 21 instead of JDK 17 for building and testing. This ensures compatibility with newer Java features and aligns with updated project requirements.

Ludy87 committed:
Update JDK version to 21 in tauri-build workflow

The GitHub Actions workflow now sets up JDK 21 instead of JDK 17 to ensure compatibility with newer Java features and dependencies.

Ludy87 committed:
Update dependencies and CI build matrix

Updated several Rust and JavaScript dependencies to their latest versions, including @tauri-apps/api, and made corresponding changes in Cargo.lock and package-lock.json. The Tauri build GitHub Actions workflow was also updated to remove duplicate macOS build targets and streamline the build matrix.

Ludy87 committed:
Enable native keyring features for Apple and Windows

Updated the keyring dependency in Cargo.toml to enable 'apple-native' and 'windows-native' features, improving native integration on those platforms. Cargo.lock updated to reflect new and updated dependencies.

Ludy87 committed:
Merge branch 'main' into AppImage_fix

Frooodle committed:
Merge branch 'main' into AppImage_fix

Ludy87 committed:
Add create-dmg installation step for macOS builds

Adds a step to install the 'create-dmg' tool via Homebrew when building on macOS platforms. This ensures DMG creation tooling is available for packaging macOS builds.

Ludy87 committed:
Enhance multiOS and Tauri build workflows

Adds macOS DMG tooling installation and dependency checks to multiOSReleases workflow. Improves artifact handling for RPM packages and unknown platforms. Updates Tauri build workflow for consistency in formatting and dependency installation steps.

Ludy87 committed:
Merge branch 'main' into AppImage_fix

Frooodle commented on frontend/src-tauri/Cargo.toml:
```diff
@@ -3,8 +3,10 @@ name = "stirling-pdf"
 authors = ["Stirling-PDF Contributors"]
-license = ""
-repository = ""
+license = "MIT"
```
I think we can do without for now... yolo :')

Ludy87 committed:
Merge branch 'main' into AppImage_fix

Ludy87 committed:
Remove license field from Cargo.toml

Cleared the license field in the Cargo.toml for the Stirling-PDF desktop application. This may be to update or clarify the project's licensing information.

Ludy87 committed:
Merge branch 'main' into AppImage_fix

Ludy87 committed:
Update workflows and add new frontend dependencies

Ludy87 committed:
Merge branch 'main' into AppImage_fix

Ludy87 committed:
Update tauri.conf.json

Ludy87 committed:
Merge branch 'main' into AppImage_fix
